### PR TITLE
Callib3d 4channel image detection

### LINKED ISSUE #8326: Bug report:
image

New code 3.2 version does not support 4 channel color image detections. 3.1 does.

### PR DESCRIPTION
**Pull request by ottogin**
3.2 version doesn't support 4 channel color image detections, unlike 3.1. Now, we don't call
CV_Error() if a 4 channel image is given. And process it, using COLOR_BGRA2GRAY flag

resolve #8326

I am doubting COLOR_RGBA2GRAY flag is needed instead of COLOR_BGRA2GRAY?

### CODE REVIEW

File: modules/calib3d/src/calibinit.cpp [hunk 1]
```diff
@@ -451,11 +451,16 @@ int cvFindChessboardCorners( const void* arr, CvSize pattern_size,
     if( !out_corners )
         CV_Error( CV_StsNullPtr, "Null pointer to corners" );
 
-    if (img.channels() != 1)
+    if (img.channels() == 3)
     {
         cvtColor(img, img, COLOR_BGR2GRAY);
```
Comment (alalek) [inline modules/calib3d/src/calibinit.cpp]: COLOR_BGR2GRAY is enough. It works with both 3 and 4 channel images (like COLOR_BGRA2GRAY).
So this change is not needed at all.

Please eliminate unrelated commits too (`git rebase -i HEAD~2` should help).

Comment (ottogin) [inline modules/calib3d/src/calibinit.cpp]: Why don't we process with COLOR_RGB2GRAY?

Comment (alalek) [inline modules/calib3d/src/calibinit.cpp]: COLOR_RGB2GRAY and COLOR_BGR2GRAY are not equivalent, so changing them doesn't work.
BGR is the current color format for imread/imshow functions (on Windows/Linux/Mac).
RGBA is used on Android for camera capture, so be careful. There is some issue with "description" about this.
